File: challenge82.py
import sys

test_cases = open(sys.argv[1], 'r')

for test in test_cases:
    # Remove new line characters
    test = test.strip('\n')

    # Convert to list of integers
    nums = [int(num) for num in test]

    # Determine if number is an Armstring number
    print int(test) == sum([num ** len(nums) for num in nums])

test_cases.close()

File: Ruby/challenge82.rb
File.open(ARGV[0]).each_line do |line|
  digits = line.strip().split(//)
  # Determine if number is an Armstrong number
  sum = digits.inject(0) { |sum, n| sum + n.to_i ** digits.length }
  # Print 'True' or 'False' based on result
  if sum == line.to_i then puts 'True' else puts 'False' end
end
